src/goto-programs/goto_convert.cpp

```diff
@@ -199,7 +199,8 @@ void goto_convertt::finish_gotos(goto_programt &dest)
             i.code.add_source_location(),
             unwind_to_size,
             destructor_code,
-            goto_stack);
+            goto_stack,
+            symbol_table.lookup(i.function)->mode);
```

Comment:
Nit pick: could we use `ns.lookup(i.function).mode` here?

Reply:
If you end up leaving as it is, use `lookup_ref` since immediately dereferencing

Reply:
Changin this to `ns.lookup(i.function).mode` or to `ns.lookup_ref(i.function).mode` make most of the tests in cbmc-java fail (with an identifier not found exception in the first case and core dumped in the second case)

Reply:
I'm taking a look.

Reply:
@tautschnig I removed this lookup and just used `mode` coming from the calling function.

Reply:
Ok, makes sense! I'll still invest some time to debug the reasons for the problem as it could be useful otherwise, but your approach is a very good one to unblock things.

Reply:
Ok, `finish_gotos` will never be able to use `i.function`, because that field is only set later from `goto_convert_functionst::convert_function`. The chosen approach is probably the only correct one.